stablehlo/transforms/StablehloLegalizeDeprecatedOps.cpp

```diff
@@ -0,0 +1,162 @@
+/* Copyright 2023 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// Implements composite inlining.
+
+#include <cassert>
+#include <cstdint>
+
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/Location.h"
+#include "mlir/IR/PatternMatch.h"
+#include "stablehlo/dialect/StablehloOps.h"
+#include "stablehlo/transforms/Passes.h"
+
+namespace mlir {
+namespace stablehlo {
+
+#define GEN_PASS_DEF_STABLEHLOLEGALIZEDEPRECATEDOPSPASS
+#include "stablehlo/transforms/Passes.h.inc"
+
+namespace {
+
+struct StablehloLegalizeDeprecatedOpsPass final
+    : impl::StablehloLegalizeDeprecatedOpsPassBase<
+          StablehloLegalizeDeprecatedOpsPass> {
+  StablehloLegalizeDeprecatedOpsPass()
+      : StablehloLegalizeDeprecatedOpsPassBase<
+            StablehloLegalizeDeprecatedOpsPass>() {};
+  StablehloLegalizeDeprecatedOpsPass(
+      const StablehloLegalizeDeprecatedOpsPassOptions &opts)
+      : StablehloLegalizeDeprecatedOpsPassBase<
+            StablehloLegalizeDeprecatedOpsPass>(opts) {};
```

Comment:
I believe `using StablehloLegalizeDeprecatedOpsPassBase::StablehloLegalizeDeprecatedOpsPassBase` should bring these in?

Reply:
Nice, didn't realize that.